File: rotate_matrix_90/rotate_matrix_90.py
def rotate(matrix):
    """
    :type matrix: List[List[int]]
    :rtype: None Do not return anything, modify matrix in-place instead.
    """
    for i in range(len(matrix)):
        for j in range(i):
            temp = matrix[i][j]
            matrix[i][j] = matrix[j][i]
            matrix[j][i] = temp
    for i in range(len(matrix)):
        matrix[i] = matrix[i][::-1]
    return matrix

print(rotate([[1,2,3], [4,5,6], [7,8,9]]))

File: rotate_matrix_90/rotate_matrix_90.cpp
#include<bits/stdc++.h>

using namespace std;

void rotate(vector<vector<int>>& matrix) {
        for (int i=0; i<matrix.size(); i++){
            for (int j=0; j<i; j++){
                swap(matrix[i][j], matrix[j][i]);
            }
        
        }
        for (int i=0; i<matrix.size(); i++){
            reverse(matrix[i].begin(), matrix[i].end());
        }
        
}

int main(){
    vector<vector<int>> a{{1,2,3}, {4,5,6}, {7,8,9}};
    rotate(a);
    for (int i = 0; i < a.size(); i++) {
    for (int j = 0; j < a[0].size(); j++) {
      cout << a[i][j] << " ";
    }
    cout << "\n";
  }
}